src/odeproblem.cpp

```diff
@@ -380,6 +380,12 @@ void odeproblem::advan2(const double& tfrom, const double& tto) {
   
   if(k10 <= 0) Rcpp::stop("k10 has a 0 or negative value");
   
+  // Going for the simpler method; using epsilon of ~  100 * DBL_EPSILON
+  // https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/
+  if(std::fabs(ka-k10) < 2.22e-14) {
```

Comment:
I think that approach makes sense here.

---

Just a note:`isSymmetric.matrix` uses the `100 * DBL_EPSILON` as its default value.

https://github.com/wch/r-source/blob/9d01d16b4cda7b53b9a2c94a1e3f078faf1b9eaa/src/library/base/R/eigen.R#L22